File: frontend/src/api/analysis.ts
import api from './client'

export interface RunAnalysisRequest {
  resume_id: string
  jd_id: string
}

export interface AnalysisStatus {
  run_id: string
  status: 'pending' | 'running' | 'completed' | 'failed' | 'partial'
  current_phase?: string
  progress_pct?: number
}

export interface CategoryScore {
  category: string
  score: number
  weight: number
  reasoning: string
  matched_items: string[]
  missing_items: string[]
}

export interface ScoreOutput {
  overall_score: number
  category_scores: CategoryScore[]
  matched_keywords: string[]
  missing_keywords: string[]
  skills_gap: string[]
  certification_gap: string[]
  achievement_gap: string[]
}

export interface AnalysisResult {
  run_id: string
  status: string
  pass1_score: ScoreOutput | null
  pass2_score: ScoreOutput | null
  gap_report: any
  ats_report: any
  parsed_resume: any
  tailored_resume: any
  cover_letter: any
  interview_guide: any
  total_tokens_used: number
  total_cost_usd: number
}

export async function startAnalysis(data: RunAnalysisRequest): Promise<{ run_id: string }> {
  const res = await api.post('/analysis/run', data)
  return res.data
}

export async function getAnalysisStatus(runId: string): Promise<AnalysisStatus> {
  const res = await api.get(`/analysis/${runId}/status`)
  return res.data
}

export async function getAnalysisResult(runId: string): Promise<AnalysisResult> {
  const res = await api.get(`/analysis/${runId}`)
  return res.data
}

File: backend/app/schemas/analysis.py
"""Analysis run request/response schemas."""

import uuid
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel, Field


class AnalysisRunRequest(BaseModel):
    """Request to initiate a new analysis pipeline run."""

    resume_id: uuid.UUID
    jd_id: uuid.UUID


class AnalysisStatusResponse(BaseModel):
    """Lightweight status response for polling."""

    run_id: uuid.UUID
    status: str = Field(
        description="One of: pending, running, completed, failed, partial."
    )
    current_phase: Optional[str] = None
    progress_pct: int = Field(ge=0, le=100, default=0)


class AnalysisResultResponse(BaseModel):
    """Full analysis result including all agent outputs."""

    run_id: uuid.UUID
    status: str
    pass1_score: Optional[dict[str, Any]] = None
    pass2_score: Optional[dict[str, Any]] = None
    gap_report: Optional[dict[str, Any]] = None
    ats_report: Optional[dict[str, Any]] = None
    parsed_resume: Optional[dict[str, Any]] = None
    parsed_jd: Optional[dict[str, Any]] = None
    tailored_resume: Optional[dict[str, Any]] = None
    verification_report: Optional[dict[str, Any]] = None
    cover_letter: Optional[dict[str, Any]] = None
    interview_guide: Optional[dict[str, Any]] = None
    total_tokens_used: int = 0
    total_cost_usd: float = 0.0
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    created_at: datetime

    model_config = {"from_attributes": True}
